### PR TITLE
 Set if the bottom and top should have a lid 

### PR DESCRIPTION
**Pull request by thuijzer**
Sometimes you don't want a bottom or top lid/cap.
Added two optional boolean options: bottomLid and topLid so you can turn off the lid.

### CODE REVIEW

File: src/geometries/ExtrudeGeometry.js [hunk 1]
```diff
@@ -565,36 +574,51 @@ function ExtrudeBufferGeometry( shapes, options ) {
 
 				// Top faces
 
-				for ( i = 0; i < flen; i ++ ) {
+				if ( topLid ) {
```
Comment (WestLangley) [inline src/geometries/ExtrudeGeometry.js]: formatting: add blank line

If this merged, please also update the docs.

File: examples/webgl_geometry_spline_editor.html [hunk 2]
```diff
@@ -151,7 +151,7 @@
 
 				controls.addEventListener( 'start', function () {
 
-					cancelHideTransorm();
+					cancelHideTransform();
```
Comment (WestLangley) [inline examples/webgl_geometry_spline_editor.html]: Shouldn't the changes in this file be in a separate PR?

**Comment by Mugen87** [comment]
Um, `CylinderBufferGeometry` has `openEnded` which defines whether the top/bottom segments of the cylinder are opened or not. I would prefer a consistent approach and not something special for each geometry generator.

**Comment by WestLangley** [comment]
>I would prefer a consistent approach and not something special for each geometry generator.

This is about `ExtrudeGeometry`, not `CylinderGeometry`. Boxes also have tops and bottoms. I would not worry about it.

You can add the same options to `CylinderGeometry` if you want. I prefer `options` to a long list of args, anyway: it's much easier to change the API.

Plus, this new feature is hidden in the `options` and is backwards compatible.

**Comment by Mugen87** [comment]
Closing for now. Let's keep `ExtrudeGeometry` as it is since the class is already the most complex geometry generator. If @mrdoob thinks it's still worth, we can reopen and merge the PR. In this case, I'll update the docs and TS file (unless the OP does this in the meanwhile).
